implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalfxDistributionSummary.java

```diff
@@ -16,88 +16,61 @@
 package io.micrometer.signalfx;
 
 import io.micrometer.common.lang.Nullable;
-import io.micrometer.core.instrument.AbstractDistributionSummary;
 import io.micrometer.core.instrument.Clock;
 import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
 import io.micrometer.core.instrument.distribution.HistogramSnapshot;
-import io.micrometer.core.instrument.distribution.TimeWindowMax;
-import io.micrometer.core.instrument.step.StepTuple2;
-
-import java.util.concurrent.atomic.DoubleAdder;
-import java.util.concurrent.atomic.LongAdder;
+import io.micrometer.core.instrument.distribution.StepBucketHistogram;
+import io.micrometer.core.instrument.step.StepDistributionSummary;
 
 /**
- * This class is mostly the same as
- * {@link io.micrometer.core.instrument.step.StepDistributionSummary}, with one notable
- * difference: the {@link DistributionStatisticConfig} is modified before being passed to
- * the super class constructor - that forces the histogram generated by this meter to be
- * cumulative.
+ * A StepDistributionSummary which provides support for multiple flavours of Histograms to
+ * be recorded based on {@link SignalFxConfig#publishCumulativeHistogram()} and
+ * {@link SignalFxConfig#publishDeltaHistogram()}.
  *
  * @author Bogdan Drutu
  * @author Mateusz Rzeszutek
+ * @author Lenin Jaganathan
  */
-final class SignalfxDistributionSummary extends AbstractDistributionSummary {
-
-    private final LongAdder count = new LongAdder();
-
-    private final DoubleAdder total = new DoubleAdder();
-
-    private final StepTuple2<Long, Double> countTotal;
-
-    private final TimeWindowMax max;
+final class SignalfxDistributionSummary extends StepDistributionSummary {
```

Comment:
This makes SignaFxMeters simple and makes them closely aligns with Step contracts. The earlier attempt to use AbstractDistributionSummary was due to the lack of flexibility on Histogram which was fixed in 1.11.0 by adding a new constructor which takes Histogram rather than DistributionStatistic Config.

Reply:
A positive side effect of using the StepMeters in StepRegistry might fix https://github.com/micrometer-metrics/micrometer/issues/3845.

I guess this fixes https://github.com/micrometer-metrics/micrometer/issues/3845.